components/brave_wallet/browser/internal/hd_key_ed25519_unittest.cc

```diff
@@ -4,107 +4,107 @@
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 #include "brave/components/brave_wallet/browser/internal/hd_key_ed25519.h"
+
+#include <vector>
+
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/string_util.h"
+#include "brave/components/brave_wallet/common/hex_utils.h"
 #include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/boringssl/src/include/openssl/curve25519.h"
 
 namespace brave_wallet {
 
+namespace {
+
+bool VerifySignature(const HDKeyEd25519& key,
+                     base::span<const uint8_t> msg,
+                     base::span<const uint8_t, kEd25519SignatureSize> sig) {
+  return !!ED25519_verify(msg.data(), msg.size(), sig.data(),
+                          key.GetPublicKeyAsSpan().data());
+}
+
+}  // namespace
+
 // https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-1-for-ed25519
 TEST(HDKeyEd25519UnitTest, TestVector1) {
   std::vector<uint8_t> bytes;
   EXPECT_TRUE(
       base::HexStringToBytes("000102030405060708090a0b0c0d0e0f", &bytes));
 
   // m
-  auto master_key_base = HDKeyEd25519::GenerateFromSeed(bytes);
-  EXPECT_EQ(master_key_base->GetPath(), "m");
-  HDKeyEd25519* master_key = static_cast<HDKeyEd25519*>(master_key_base.get());
-  EXPECT_EQ(
-      base::ToLowerASCII(base::HexEncode(master_key->GetPrivateKeyBytes())),
-      "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19e7");
-  EXPECT_EQ(
-      base::ToLowerASCII(base::HexEncode(master_key->GetPublicKeyBytes())),
-      "a4b2856bfec510abab89753fac1ac0e1112364e7d250545963f135f2a33188ed");
+  auto master_key = HDKeyEd25519::GenerateFromSeedAndPath(bytes, "m");
+  EXPECT_EQ(HexEncodeLower(master_key->GetPrivateKeyAsSpan()),
```

Comment:
I think it would be better if you didn't make this change to use `HexEncodeLower` in this PR. It makes it harder to see that everything is still the same in terms of expectations. Not critical, but it would be an easier read and you could follow-up to change the tests

Reply:
rolled back